packages/engine/core/src/lib/abstract.ts

```diff
@@ -77,48 +77,39 @@ export class Plugin<T extends Api = any, App extends ApiMap = any> implements Pl
     return this[method](...args)
   }
 
+  protected setCurrentRequest(request: PluginRequest) {
+    this.currentRequest = request
+  }
+
+  protected letContinue() {
+    delete this.currentRequest
+    this.queue = this.queue.filter((value) => {
+      return value.canceled === false && value.timedout === false && value.finished === false
+    })
+    const next = this.queue.find((value) => {
+      return value.canceled === false && value.timedout === false && value.finished === false
+    })
+    if (next) next.run()
+  }
+
   /** Add a request to the list of current requests */
   protected addRequest(request: PluginRequest, method: Profile<T>['methods'][number], args: any[]) {
     return new Promise((resolve, reject) => {
-      // Add a new request to the queue
-      this.requestQueue.push(async () => {
```

Comment:
this can be removed:
`protected currentRequest: PluginRequest`

Reply:
not true, it is set
```
  protected setCurrentRequest(request: PluginRequest) {
    this.currentRequest = request
  }
```